internal/aggregator/v1alpha1/v1alpha_aggregator.go

```diff
@@ -0,0 +1,86 @@
+package v1alpha1
+
+import (
+	"sync"
+
+	"github.com/talos-systems/os-engine/pkg/sink"
+	"github.com/talos-systems/os-engine/pkg/source"
+)
+
+type Aggregator struct {
+	sources []source.Source
+	sinks   []sink.Sink
+}
+
+func New() *Aggregator {
+	return &Aggregator{}
+}
+
+func (a *Aggregator) Start(sinks []sink.Sink, sources []source.Source) error {
+	a.sources = sources
+	a.sinks = sinks
+
+	var wg sync.WaitGroup
+
+	wg.Add(len(a.sources))
+
+	for _, source := range a.sources {
+		source := source
+
+		err := source.Load()
+		if err != nil {
+			return err
+		}
+
+		go func() {
+			for msg := range source.Recv() {
+				for _, sink := range a.sinks {
+					go sink.Send(msg)
```

Comment:
feels like unbound goroutine creation... this could be a problem

Reply:
I plan on getting the event system you wrote for Talos and replacing this whole implementation. This  whole aggregator thing was thrown together for the poc. Actually, that is what I wanted to chat with you about today.

Reply:
Updated to use the event interface from Talos.